diesel_tests/tests/types.rs

```diff
@@ -1464,3 +1464,52 @@ fn test_inserting_ranges() {
     assert_eq!(v1, Some(value));
     assert_eq!(v2, value);
 }
+
+#[test]
+#[cfg(feature = "postgres")]
+fn char_from_sql() {
+    assert_eq!('a', query_single_value::<CChar, char>(r#"'a'::"char""#));
+    assert_eq!(' ', query_single_value::<CChar, char>(r#"' '::"char""#));
+    assert_eq!('~', query_single_value::<CChar, char>(r#"'~'::"char""#));
+    assert_eq!('5', query_single_value::<CChar, char>(r#"'5'::"char""#));
+    assert_eq!('\\', query_single_value::<CChar, char>(r#"'\'::"char""#));
+    assert_eq!('\"', query_single_value::<CChar, char>(r#"'"'::"char""#));
+    assert_eq!('\'', query_single_value::<CChar, char>(r#"''''::"char""#));
+    assert_eq!('`', query_single_value::<CChar, char>(r#"'`'::"char""#));
+    assert_eq!(
+        char::from(195),
+        query_single_value::<CChar, char>(r#"'ö'::"char""#)
+    );
+    assert_eq!(
+        char::from(195),
+        query_single_value::<CChar, char>(r#"'ä'::"char""#)
+    );
+    assert_eq!('\0', query_single_value::<CChar, char>(r#"0::"char""#));
+    assert_eq!('0', query_single_value::<CChar, char>(r#"'0'::"char""#));
+}
+
+#[test]
+#[cfg(feature = "postgres")]
+fn char_to_sql() {
+    assert!(query_to_sql_equality::<CChar, char>(r#"'a'::"char""#, 'a'));
+    assert!(query_to_sql_equality::<CChar, char>(r#"' '::"char""#, ' '));
+    assert!(query_to_sql_equality::<CChar, char>(r#"'~'::"char""#, '~'));
+    assert!(query_to_sql_equality::<CChar, char>(r#"'5'::"char""#, '5'));
+    assert!(query_to_sql_equality::<CChar, char>(r#"'\'::"char""#, '\\'));
+    assert!(query_to_sql_equality::<CChar, char>(r#"'"'::"char""#, '\"'));
+    assert!(query_to_sql_equality::<CChar, char>(
+        r#"''''::"char""#,
+        '\''
+    ));
+    assert!(query_to_sql_equality::<CChar, char>(r#"'`'::"char""#, '`'));
+    assert!(query_to_sql_equality::<CChar, char>(
+        r#"'ö'::"char""#,
+        char::from(195)
+    ));
+    assert!(query_to_sql_equality::<CChar, char>(
+        r#"'ä'::"char""#,
+        char::from(195)
+    ));
+    assert!(query_to_sql_equality::<CChar, char>(r#"0::"char""#, '\0'));
+    assert!(query_to_sql_equality::<CChar, char>(r#"'0'::"char""#, '0'));
```

Comment:
Add a test here, to show that it errors for something that's not ASCII

Reply:
I had another though on this. If a PostgreSQL char is always represented by whatever fits in one byte, we should maybe just use the corresponding rust type for that. That would be `u8` and not `char`. This would move the check what's valid or not to the compiler, while it also allows to use char literals (via the `b'…'` modifier.). See [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6cdc4c6f2e80bbeeb0830749ed139684) for some simple demonstration. 
I'm sorry that this idea only came up now, but I would prefer using `u8` instead of `char` there as this much closer matches what's postgresql supports there.